server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsService.java

```diff
@@ -147,33 +148,30 @@ public void addClientDetails(ClientDetails clientDetails, String zoneId) throws
 
     private boolean exists(String clientId, String zoneId) {
         List<String> idResults = jdbcTemplate.queryForList(SINGLE_SELECT_STATEMENT, String.class, clientId, zoneId);
-        if (idResults != null && idResults.size() == 1) {
-            return true;
-        }
-        return false;
+        return idResults.size() == 1;
```

Comment:
Don't we need the null check still? Or is it guaranteed that `idResults` will never be null?

Reply:
sonar says it is never null